dask/dataframe/tests/test_dataframe.py

```diff
@@ -936,7 +940,11 @@ def f(df, partition_info=None):
         assert dsk[("x", d.divisions.index(partition_info["division"]))].equals(df)
         return df
 
-    result = d.map_partitions(f, meta=d).compute(scheduler="single-threaded")
+    df = d.map_partitions(f, meta=d)
+    layer = [layer for k, layer in df.dask.layers.items() if k.startswith("f-")][0]
```

Comment:
Not meant as a blocking comment, but this should be equivalent and a nice opportunity to use the `hlg_layer` helper you added recently 

```suggestion
    layer = hlg_layer(df.dask, "f-")
```

Reply:
Good catch, I thought I'd marked them with comments but clearly missed this one